From fd942187934a39260f1a50c8e22bc36108061f4f Mon Sep 17 00:00:00 2001
From: Christian Persch <chpe@src.gnome.org>
Date: Sun, 24 Jan 2021 20:49:49 +0100
Subject: [PATCH] Revert "lib: Sanitise the passed-in font description"

This needed more work which is done on master but too involved for stable.
---
 src/vte.cc | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/src/vte.cc b/src/vte.cc
index f6d0a664..5a631d96 100644
--- a/src/vte.cc
+++ b/src/vte.cc
@@ -7340,14 +7340,6 @@ Terminal::set_font_desc(PangoFontDescription const* font_desc)
 				"Using default monospace font.\n");
 	}
 
-        /* Sanitise the  font description. Style and weight need to be default here,
-         * since those are set via SGR attributes; and gravity makes no sense in vte.
-         */
-        pango_font_description_unset_fields(desc,
-                                            PangoFontMask(PANGO_FONT_MASK_GRAVITY |
-                                                          PANGO_FONT_MASK_STYLE |
-                                                          PANGO_FONT_MASK_WEIGHT));
-
         bool const same_desc = m_unscaled_font_desc &&
                 pango_font_description_equal(m_unscaled_font_desc.get(), desc);
 
-- 
GitLab

